tests/unit/test_client.py

```diff
@@ -16,6 +22,51 @@
 from tests.unit.test_thrift_backend import ThriftBackendTestSuite
 from tests.unit.test_arrow_queue import ArrowQueueSuite
 
+class ThriftBackendMockFactory:
+
+    @classmethod
+    def new(cls):
+        ThriftBackendMock = Mock(spec=ThriftBackend)
+        ThriftBackendMock.return_value = ThriftBackendMock
```

Comment:
What?  A mock that returns itself when invoked?

Reply:
Not quite. This is a mock factory. When you call the `new()` method, it builds a fresh mock and hands it to you.

Originally I just declared the mock once at the top of the file. But there's one test that makes assertions about method call counts, and since the same mock was only instantiated once, that test _passed_ when I ran it alone and _failed_ when I ran the whole suite. So I wrapped it all into a factory and called it a day.

Didn't want to spend too much time on this since I'm confident these are failures due to API changes (that's why I had to start applying `PropertyMock` patches where they weren't previously required.

Reply:
I mean specifically you assigning a Mock to the label `ThriftBackendMock`, and then:
```
ThriftBackendMock.return_value = ThriftBackendMock
```

How does that not make it its own return value?

Reply:
oh whoops, I misunderstood your comment (didn't see the context at first)

Yeah, I thought this was strange as well. Basically our `client` isn't consistent about whether it wants an instantiated `ThriftBackend` or not (I'll be fixing that in the next few weeks). And our `ThriftBackend` isn't a "good" Python object in the sense that it has a bunch of properties that are defined at runtime but can't be inspected by the `unittest.mock` library before running an individual test.

In a perfect world, we should be able to call `MagicMock(autospec=ThriftBackend)` and it will give us a mock that, when called, returns an "instantiated" instance of itself. But autospec won't work because of those unpublished properties. So I had to code that logic myself on this line.

Reply:
tl;dr if we don't include this line, calling `ThriftBackendMock()` will return a new, generic `Mock()` object rather than an instance of our true mock. So I include this logic to make sure that whether you call `ThriftBackend.execute_command()` or `ThriftBackend().execute_command()` you still get the `MockTExecuteResponse()`.